### ISSUE TITLE
Minify Asciidoctor.js using the advanced compilation mode (Google Clojure Compiler)

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor.js#723]
Currently Asciidoctor.js is minified using Google Clojure Compiler in simple mode.

The advanced compilation mode requires a few changes to Opal.
For instance, we should use quoted string to refer to object properties: https://developers.google.com/closure/compiler/docs/limitations

```
-$defineProperty(klass.$$prototype, '$$class', klass);
+$defineProperty(klass['$$prototype'], '$$class', klass);
```

https://github.com/opal/opal/blob/a1f36da26a2da31770da3f0e24ea1ddd4fd4e7a1/opal/corelib/runtime.js#L439

It's possible to annotate the code to give hints: https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler

Since the code is altered, we need to run the tests suite on the produced file to make sure that Asciidoctor.js is fully working.

**Comment by mojavelinux** [comment]
Agreed. It would be great if there were a switch to allow the test suite to run on either the unminified or minified code. That way, it doesn't slow down development at all.
